community/modules/scripts/spack-install/templates/spack_setup.yml.tpl

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Install Application
+  hosts: localhost
+  vars:
+    app_name: ${app_name}
+    profile_script: ${profile_script}
+    install_dir: ${install_dir}
+    git_url: ${git_url}
+    git_ref: ${git_ref}
+    chmod_mode: ${chmod_mode}
+    chown_owner: ${chown_owner}
+    chgrp_group: ${chgrp_group}
+    finalize_setup_script: ${finalize_setup_script}
+  tasks:
+  - name: Print Application Name
+    ansible.builtin.debug:
+      msg: "Running installation for application: {{app_name}}"
+
+  - name: Add profile script for application
+    ansible.builtin.copy:
+      dest: /etc/profile.d/{{ app_name }}.sh
+      mode: '0644'
+      content: "{{ profile_script }}"
+
+  - name: Create parent of install directory
+    ansible.builtin.file:
+      path: "{{ install_dir | dirname }}"
+      state: directory
+
+  - name: Acquire lock
+    ansible.builtin.command:
+      mkdir "{{ install_dir | dirname }}/.install_{{ app_name }}_lock"
+    register: lock_out
+    changed_when: lock_out.rc == 0
+    failed_when: false
+
+  - name: Clones into installation directory
+    ansible.builtin.command: git clone --branch {{ git_ref }} {{ git_url }} {{ install_dir }}
+    when: lock_out.rc == 0
+
+  - name: chgrp on installation
```

Comment:
Please avoid the clumsiness here by combining these into a single task using [filters](https://docs.ansible.com/ansible/2.9/user_guide/playbooks_filters.html#omitting-parameters); note setting the 2nd parameter to true so that it evaluates upon the Falsey empty string.

https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-filters.default

```
    ansible.builtin.file:
      path: "{{ install_dir }}"
      owner: "{{ chown_owner | default(omit, true) }}"
      group: "{{ chgrp_group | default(omit, true)  }}"
      mode: "{{ chmod_mode | default(omit, true)  }}"
      recurse: true
    when:  lock_out.rc == 0
```

FYI: I believe comparison to empty string would trigger an opt-in linting error.

https://ansible.readthedocs.io/projects/lint/rules/empty-string-compare/